### PR TITLE
Use XorK bloq instead of CNOTs to load data in QROM

### PR DESCRIPTION
**Pull request by tanujkhattar**
This improves the readability of the decomposition and reduces the number of subbloqs in the resulting composite bloq. Should also help with https://github.com/quantumlib/Qualtran/issues/1052

See old and new diagrams generated by the following test code
```python
data1 = np.arange(10)
data2 = np.arange(10) * 100
qrom = QROM.build_from_data(data1, data2)
show_bloq(qrom.decompose_bloq(), 'latex')
```

Old Diagram:
qrom_old

New Diagram:
qrom_new

### CODE REVIEW

**Comment by fdmalone** [comment]
NIIIICE

File: qualtran/bloqs/data_loading/qrom.py [hunk 1]
```diff
@@ -136,20 +137,21 @@ def _load_nth_data(
             assert all(isinstance(x, (int, numbers.Integral)) for x in target_shape)
             for idx in np.ndindex(cast(Tuple[int, ...], target_shape)):
                 data_to_load = int(d[selection_idx + idx])
-                for q, bit in zip(target[idx], f'{data_to_load:0{target_bitsize}b}'):
-                    if int(bit):
-                        yield gate(q)
+                yield XorK(QUInt(target_bitsize), data_to_load).on(*target[idx]).controlled_by(
+                    *ctrl_qubits
+                )
+                # for q, bit in zip(target[idx], f'{data_to_load:0{target_bitsize}b}'):
```
Comment (fdmalone) [inline qualtran/bloqs/data_loading/qrom.py:143]: remove commented out code? OR say equivalent to if it's helpful for readability

**Review by fdmalone — APPROVED** [review]

File: qualtran/bloqs/data_loading/qrom.py (hunk 1 above)
Comment (tanujkhattar) [reply qualtran/bloqs/data_loading/qrom.py:143]: Had to remove, forgot. Had commented out assuming there would be test failures that I may need to go and debug; fortunately; there were no test failures :)

**Comment by mpharrigan** [comment]
niiiice

**Pull request merged**